src/utils/optionsMatches.js

```diff
@@ -0,0 +1,42 @@
+import matchesStringOrRegExp from "./matchesStringOrRegExp"
+
+const extractValueFromInput = (property, input) => {
```

Comment:
> If mitigating this is more of a priority than avoiding code duplication

I think so. I think the problem with this approach is that it creates a fragile abstraction that doesn't solve a difficult problem. If have a bunch of rules accessing their value on `node.value`, I don't consider that the bad kind of code duplication.